stdlib/join/equijoin_test.go

```diff
@@ -0,0 +1,309 @@
+					plan.CreateLogicalNode("from2", &influxdb.FromProcedureSpec{}),
+					plan.CreateLogicalNode("filter3", &universe.FilterProcedureSpec{}),
+					plan.CreatePhysicalNode("join.join4", &join.EquiJoinProcedureSpec{}),
+				},
+				Edges: [][2]int{
+					{0, 1},
+					{2, 3},
+					{1, 4},
+					{3, 4},
+				},
+				Now: now,
+			},
+		},
+		{
+			name: "reject or",
+			flux: `import "join"
+			left = from(bucket: "b1", host: "http://localhost:8086")
+				|> filter(fn: (r) => r._measurement == "a")
+			right = from(bucket: "b2", host: "http://localhost:8086")
+				|> filter(fn: (r) => r._measurement == "b")
+			join.join(
+				left: left,
+				right: right,
+				on: (l, r) => l.a == r.b or l.e == r.f,
+				as: (l, r) => ({l with c: r._value}),
+				method: "inner",
+			)`,
+			wantErr: errors.New(codes.Invalid, "unsupported operator in join predicate: or"),
+		},
+		{
+			name: "reject intratable comparisons",
+			flux: `import "join"
+			left = from(bucket: "b1", host: "http://localhost:8086")
+				|> filter(fn: (r) => r._measurement == "a")
+			right = from(bucket: "b2", host: "http://localhost:8086")
+				|> filter(fn: (r) => r._measurement == "b")
+			join.join(
+				left: left,
+				right: right,
+				on: (l, r) => l.a == l.b,
+				as: (l, r) => ({l with c: r._value}),
+				method: "inner",
+			)`,
+			wantErr: errors.New(
+				codes.Invalid,
+				"binary expression operands must reference `l` or `r` only, and may not reference the same object",
+			),
+		},
+		{
+			name: "r == l",
+			flux: `import "join"
+			left = from(bucket: "b1", host: "http://localhost:8086")
+				|> filter(fn: (r) => r._measurement == "a")
+			right = from(bucket: "b2", host: "http://localhost:8086")
+				|> filter(fn: (r) => r._measurement == "b")
+			join.join(
+				left: left,
+				right: right,
+				on: (l, r) => r.a == l.b,
+				as: (l, r) => ({l with c: r._value}),
+				method: "inner",
+			)`,
+			wantPairs: []join.ColumnPair{
+				join.ColumnPair{Left: "b", Right: "a"},
+			},
+			wantPlan: &plantest.PlanSpec{
+				Nodes: []plan.Node{
+					plan.CreateLogicalNode("from0", &influxdb.FromProcedureSpec{}),
+					plan.CreateLogicalNode("filter1", &universe.FilterProcedureSpec{}),
+					plan.CreateLogicalNode("from2", &influxdb.FromProcedureSpec{}),
+					plan.CreateLogicalNode("filter3", &universe.FilterProcedureSpec{}),
+					plan.CreatePhysicalNode("join.join4", &join.EquiJoinProcedureSpec{}),
+				},
+				Edges: [][2]int{
+					{0, 1},
+					{2, 3},
+					{1, 4},
+					{3, 4},
+				},
+				Now: now,
+			},
+		},
+		{
+			name: "multiple comparison same column",
+			flux: `import "join"
+			left = from(bucket: "b1", host: "http://localhost:8086")
+				|> filter(fn: (r) => r._measurement == "a")
+			right = from(bucket: "b2", host: "http://localhost:8086")
+				|> filter(fn: (r) => r._measurement == "b")
+			join.join(
+				left: left,
+				right: right,
+				on: (l, r) => r.a == l.b and r.d == l.b,
```

Comment:
Should we explicitly test multiple comparison with an operator error for something like `on: (l, r,) => r.a == l.b and !(r.d == l.b)`?

Reply:
Yeah, that's probably a good idea. Adding a testcase for that now.